### PR TITLE
Fetch vbms_id from VACOLS if not cached

### PR DESCRIPTION
**Pull request by joofsh**
- For an appeal object that does not yet have a vbms_id, we want to try
  and fetch it from VACOLS. Once we have the vbms_id saved, we should
only reference it from the Caseflow DB

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -90,6 +90,16 @@ def saved_documents
     @saved_documents ||= fetch_documents!(save: true)
   end
 
+  # If we do not yet have the vbms_id saved in Caseflow's DB, then
+  # we want to fetch it from VACOLS, save it to the DB, then return it
+  def vbms_id
+    super || begin
```
Comment (shanear) [inline app/models/appeal.rb]: would you hate me if i said it'd be nicer if

```
begin
  check_and_load_vacols_data!
  save
  super
end
```
would be nicer as a private function like `fetch_and_save_vbms_id!`?

File: spec/models/appeal_spec.rb [hunk 2]
```diff
@@ -1125,4 +1125,28 @@
...
+
+    context "when vbms_id is nil" do
+      let(:no_vbms_id_appeal) { Appeal.create(vacols_id: appeal.vacols_id) }
+
+      it "looks up vbms_id in VACOLS" do
+        expect(no_vbms_id_appeal).to receive(:perform_vacols_request)
+          .exactly(1).times.and_call_original
+
+        expect(no_vbms_id_appeal.attributes["vbms_id"]).to be_nil
+        expect(no_vbms_id_appeal.vbms_id).to_not be_nil
```
Comment (shanear) [inline spec/models/appeal_spec.rb]: maybe `reload` here to test that it saves it as well?

File: app/models/appeal.rb (hunk 1 above)
Comment (joofsh) [inline app/models/appeal.rb]: I wouldn't hate you at all. I would in fact <3 that suggestion!

Comment (joofsh) [inline app/models/appeal.rb]: Actually sadly this won't do. Calling `super` from the private method doesn't work!

**Pull request merged**